### ISSUE TITLE
Tuning GN using inference data?

### ISSUE THREAD

**Issue by zhaohui-yang** [jiwoon-ahn/irn#16]
Dear Jiwoon, in the file 'train_irn.py', I noticed that GN was tuning using the inference data in the latest commit, [location](https://github.com/jiwoon-ahn/irn/blob/master/step/train_irn.py#L87). Is this right in the weakly supervised instance segmentation setting? I think the validation set should not be touched except for evaluation, rather than training/tuning parameters. And I'm also curious what would be affected by this? Will the mAP be improved? Thanks

**Comment by jiwoon-ahn** [comment]
1. GN is not tuned during that procedure. 
2. **"infer dataset"** is a subset of our training data, and we strictly assume no annotation except image-level class labels is available on the dataset.

What MeanShift layer does is simple: It simply collects the mean of the displacement fields from the training set, and it adjusts the range of the values at inference. The layer acts just as before. In the last commit, I just modified the layer so that it no longer depends on [BatchNorm2d](https://pytorch.org/docs/stable/nn.html#batchnorm2d).

**Comment by zhaohui-yang** [comment]
Thanks, I know what you mean. Just optimize the mean value while fixing all the other parameters, right?

**Comment by jiwoon-ahn** [comment]
You got it. :)
